fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProvider.java

```diff
@@ -34,17 +33,27 @@
 public interface SubspaceProvider {
     /**
      * This might be blocking if the subspace is never fetched before.
+     *
+     * @param context record context used to resolve the subspace.
+     *
      * @return Subspace
      */
     @Nonnull
-    Subspace getSubspace();
+    Subspace getSubspace(@Nonnull FDBRecordContext context);
 
+    /**
+     * @param context record context used to resolve the subspace.
+     *
+     * @return CompletableFuture&lt;Subspace&gt;
+     */
     @Nonnull
-    CompletableFuture<Subspace> getSubspaceAsync();
+    CompletableFuture<Subspace> getSubspaceAsync(@Nonnull FDBRecordContext context);
 
     @Nonnull
     LogMessageKeys logKey();
 
+    String toString(@Nonnull FDBRecordContext context);
```

Comment:
This should have a Javadoc making it clear what the implementor is expected / allowed to do with `context`. In particular, I think that since this is called to report errors, it should not assume that the context is working. The only non-trivial implementation just uses it as a source for the cluster key into its cache, which is safe.

Reply:
Will do.

Reply:
I guess I'll let @MMcM be the judge, but I don't think the Javadoc description actually includes the requested information about not assuming that the context is working.